src/main/java/de/tum/in/www1/artemis/web/rest/PlagiarismResource.java

```diff
@@ -1,46 +1,238 @@
+    }
+
+    /**
+     * Updates an instructor statement on a plagiarismComparison (for one side).
+     * This process will send a notification to the respective student.
+     * I.e. the instructor sets a personal message to one of the accused students.
+     *
+     * @param comparisonId the id of the PlagiarismComparison
+     * @param studentLogin of one of accused students
+     * @param statement of the instructor directed to one of the accused students
+     * @return the instructor statement (convention)
+     */
+    @PutMapping("plagiarism-comparisons/{comparisonId}/{studentLogin}/instructor-statement")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<PlagiarismStatementDTO> updatePlagiarismComparisonInstructorStatement(@PathVariable("comparisonId") long comparisonId,
+            @PathVariable("studentLogin") String studentLogin, @RequestBody PlagiarismStatementDTO statement) {
+
+        var comparison = plagiarismComparisonRepository.findByIdElseThrow(comparisonId);
+        User affectedUser = userRepository.getUserByLoginElseThrow(studentLogin);
+        Exercise affectedExercise = comparison.getPlagiarismResult().getExercise();
+        String instructorStatement = statement.statement;
+
+        if (!authenticationCheckService.isAtLeastInstructorForExercise(affectedExercise)) {
+            throw new AccessForbiddenException("Only instructors responsible for this exercise can access this plagiarism case.");
+        }
+
+        if (comparison.getSubmissionA().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonInstructorStatementA(comparison.getId(), instructorStatement);
+            comparison.setInstructorStatementA(instructorStatement); // needed for notifications
+        }
+        else if (comparison.getSubmissionB().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonInstructorStatementB(comparison.getId(), instructorStatement);
+            comparison.setInstructorStatementB(instructorStatement); // needed for notifications
+        }
+        else {
+            throw new EntityNotFoundException("Student with id not found in plagiarism comparison");
+        }
+        singleUserNotificationService.notifyUserAboutNewPossiblePlagiarismCase(comparison, affectedUser);
+        return ResponseEntity.ok(statement);
+    }
+
+    /**
+     * Retrieves the plagiarismComparison specified by its Id. The submissions are anonymized for the student.
+     * StudentIds are replaced with "Your Submission" and "Other Submission" based on the requesting user.
+     *
+     * @param comparisonId the id of the PlagiarismComparison
+     * @return the PlagiarismComparison
+     * @throws AccessForbiddenException if the requesting user is not affected by the plagiarism case.
+     */
+    @GetMapping("plagiarism-comparisons/{comparisonId}")
```

Comment:
does not follow our REST URL best practices, please improve it:
- please start with courses